From 323b2c5de2b0097566be6d58910af085b5f9da28 Mon Sep 17 00:00:00 2001
From: Lucas De Marchi <lucas.demarchi@intel.com>
Date: Fri, 2 Jun 2017 17:05:30 -0700
Subject: [PATCH 16/16] fixup! usb: otg: add cherryview support

---
 drivers/usb/dwc3/dwc3-pci.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/dwc3/dwc3-pci.c b/drivers/usb/dwc3/dwc3-pci.c
index d256064..85db4c6 100644
--- a/drivers/usb/dwc3/dwc3-pci.c
+++ b/drivers/usb/dwc3/dwc3-pci.c
@@ -160,12 +160,8 @@ static int dwc3_pci_probe(struct pci_dev *pci,
 	pdev->dev.parent = &pci->dev;
 	platform_device_add(pdev);
 
-	dwc = devm_kzalloc(dev, sizeof(*dwc), GFP_KERNEL);
-	if (!dwc)
-		return -ENOMEM;
-
-	dwc->dwc3 = platform_device_alloc("dwc3", PLATFORM_DEVID_AUTO);
-	if (!dwc->dwc3)
+	dwc3 = platform_device_alloc("dwc3", PLATFORM_DEVID_AUTO);
+	if (!dwc3) {
 		dev_err(dev, "couldn't allocate dwc3 device\n");
 		return -ENOMEM;
 	}
-- 
2.9.4

